### ISSUE TITLE
Allow to add new typedefs in module API

### ISSUE THREAD

**Issue by Totktonada** [tarantool/checkpatch#27]
We expose our types using typedefs in the module API.

Recently I got the following on PR https://github.com/tarantool/tarantool/pull/7428:

```c
ERROR: do not add new typedefs
#170: FILE: src/lib/core/decimal.h:97:
+typedef STRUCT_DECIMAL decimal_t;
```

**Comment by locker** [comment]
Module API typedefs should start with `box_` or `tt_` so this was fixed by commit 63cc5125f82641dfd2a15c75faac81d516586faa.

**Comment by Totktonada** [comment]
Thank you! It works for me.
